### ISSUE TITLE
Loss and Accuracy functions select the wrong index

### ISSUE THREAD

**Issue by ramasesh** [google-research/reverse-engineering-neural-networks#28]
The loss and accuracy functions in `utils` use the `utils.select()` function to select the logit of the RNN after processing a (variable-length) sentence.  This is *currently* done by selecting the output at the `batch['index']` time-step, but this is wrong by one.  Since `batch['index']` is equal to the length of the sentence, we actually want to select the `batch['index'] - 1` element.  

To illustrate, if the sentence is
`word0`, `word1`, `word2`,
then the network outputs will be (here `F` is the network function) 
`F(word0)`, `F(word0,word1)`, `F(word0,word1,word2)`,

So if the length of the sentence is `3`, as it is in this case, we want to select the output in position `2` (zero-indexed).  

We'll fix this by modifying the accuracy and loss functions in `utils.py`.

**Comment by ramasesh** [comment]
Was fixed in commit #30
